internal/internal_workflow_client.go

```diff
@@ -1669,55 +1734,105 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 	}, nil
 }
 
-func (w *workflowClientInterceptor) executeWorkflowWithOperation(
+func (w *workflowClientInterceptor) UpdateWithStartWorkflow(
 	ctx context.Context,
-	startRequest *workflowservice.StartWorkflowExecutionRequest,
-	operation WithStartWorkflowOperation,
-) (*workflowservice.StartWorkflowExecutionResponse, error) {
-	startOp := &workflowservice.ExecuteMultiOperationRequest_Operation{
-		Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
-			StartWorkflow: startRequest,
-		},
+	updateInput *ClientUpdateWorkflowInput,
+	startOperation WithStartWorkflowOperation,
+) (WorkflowUpdateHandle, error) {
+	startOp, ok := startOperation.(*WithStartWorkflowOperationImpl)
+	if !ok {
+		panic("startOperation must be created by NewWithStartWorkflowOperation")
 	}
 
-	var withStartOp *workflowservice.ExecuteMultiOperationRequest_Operation
-	switch t := operation.(type) {
-	case *UpdateWithStartWorkflowOperation:
-		if err := t.markExecuted(); err != nil {
-			return nil, fmt.Errorf("%w: %w", errInvalidWorkflowOperation, err)
-		}
+	// Create start request
+	if err := startOp.markExecuted(); err != nil {
+		return nil, fmt.Errorf("%w: %w", errInvalidWithStartWorkflowOperation, err)
+	}
+	if startOp.err != nil {
+		return nil, fmt.Errorf("%w: %w", errInvalidWithStartWorkflowOperation, startOp.err)
+	}
```

Comment:
What situations might this error be non-nil here? Does this check need to occur before the `markExecuted` above here?

Reply:
Thanks, I agree, it makes more sense above `markExecuted`. Changed. It is non-nil if the conflict policy is not set.